packages/node_modules/@webex/internal-plugin-devices/test/integration/spec/devices.js

```diff
@@ -342,7 +342,104 @@ describe('plugin-devices', () => {
       });
     });
 
-    describe('#getWebSocketUrl()', () => {});
+    describe('#getWebSocketUrl()', () => {
+      let services;
+
+      beforeEach('destructure services', () => {
+        services = webex.internal.services;
+      });
+
+      describe('when wait is truthy', () => {
+        let wait;
+
+        beforeEach('set wait', () => {
+          wait = true;
+        });
+
+        describe('when the device is registered', () => {
+          beforeEach('register the device', () => devices.register());
+
+          it('should resolve the promise with the websocket url',
+            () => devices.getWebSocketUrl(wait)
+              .then((url) => {
+                assert.isDefined(url);
+                assert.isTrue(services.isServiceUrl(url));
+                assert.include(url, 'mercury');
+              }));
+        });
+
+        describe('when the device is not registered', () => {
+          describe('when the device successfully registers', () => {
+            beforeEach(() => {
+              // This needs to not be returned, as it must be pending during the
+              // it clause.
+              devices.register();
+            });
+
+            it('should resolve the promise with the websocket url', () => {
+              assert.isFalse(devices.registered);
+
+              return devices.getWebSocketUrl(wait)
+                .then((url) => {
+                  assert.isDefined(url);
+                  assert.isTrue(services.isServiceUrl(url));
+                  assert.include(url, 'mercury');
+                });
+            });
+          });
+
+          describe('when the device never registers', () => {
+            it('should return a rejected promise', () => {
+              assert.isFalse(devices.registered);
+              assert.isRejected(devices.getWebSocketUrl(wait));
+            });
+          });
+        });
+      });
+
+      describe('when wait is falsy', () => {
+        let wait;
+
+        beforeEach('set wait', () => {
+          wait = false;
+        });
+
+        describe('when the device is registered', () => {
+          beforeEach('register the device', () => devices.register());
+
+          describe('when the priority host can be mapped', () => {
+            it('should resolve the promise with the websocket url', () => {
+              assert.isDefined(services.get('wdm'));
```

Comment:
There are a few of these (what I like to call) "setup assertions" throughout. They are good to help when writing tests, but might fit better in the `beforeEach` block so they don't confuse the actual test assertions.

Reply:
Reorganized these statements to assert during the beforeEach() methods to properly organize them into the "setup" stage.